wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/math/Vec3fOps.kt

```diff
@@ -0,0 +1,454 @@
+        this.z * other
+    )
+}
+
+/**
+ * Multiplies a [Vec3f] and a [Vec3d].
+ */
+public operator fun Vec3f.timesAssign(other: Vec3d) {
+    this.set(
+        this.x * other.x.toFloat(),
+        this.y * other.y.toFloat(),
+        this.z * other.z.toFloat()
+    )
+}
+
+/**
+ * Multiplies a [Vec3f] and a [Vec3f].
+ */
+public operator fun Vec3f.timesAssign(other: Vec3f) {
+    this.set(
+        this.x * other.x,
+        this.y * other.y,
+        this.z * other.z
+    )
+}
+
+/**
+ * Multiplies a [Vec3f] and a [Vec3i].
+ */
+public operator fun Vec3f.timesAssign(other: Vec3i) {
+    this.set(
+        this.x * other.x.toFloat(),
+        this.y * other.y.toFloat(),
+        this.z * other.z.toFloat()
+    )
+}
+
+/**
+ * Multiplies a [Vec3f] and a [Vector3d].
+ */
+public operator fun Vec3f.timesAssign(other: Vector3d) {
+    this.set(
+        this.x * other.x.toFloat(),
+        this.y * other.y.toFloat(),
+        this.z * other.z.toFloat()
+    )
+}
+//endregion
+
+//region Vector specific operators
+/**
+ * The dot product of a [Vec3f] and a [Vec3d].
+ */
+public infix fun Vec3f.dot(other: Vec3d): Float {
+    return this.dot(Vec3f(other))
+}
+
+/**
+ * The cross product of a [Vec3f] and a [Vec3d].
+ */
+public infix fun Vec3f.cross(other: Vec3d): Vec3f {
+    return Vec3f(
+        (this.y * other.z - this.z * other.y).toFloat(),
+        (this.z * other.x - this.x * other.z).toFloat(),
+        (this.x * other.y - this.y * other.x).toFloat()
+    )
+}
+
+/**
+ * The dot product of a [Vec3f] and a [Vec3f].
+ */
+public infix fun Vec3f.dot(other: Vec3f): Float {
+    return this.dot(other)
+}
+
+/**
+ * The cross product of a [Vec3f] and a [Vec3f].
+ */
+public infix fun Vec3f.cross(other: Vec3f): Vec3f {
+    return Vec3f(
+        this.y * other.z - this.z * other.y,
+        this.z * other.x - this.x * other.z,
+        this.x * other.y - this.y * other.x
+    )
+}
+
+/**
+ * The dot product of a [Vec3f] and a [Vec3i].
+ */
+public infix fun Vec3f.dot(other: Vec3i): Float {
+    return this.dot(Vec3f(other.x.toFloat(), other.y.toFloat(), other.z.toFloat()))
+}
+
+/**
+ * The cross product of a [Vec3f] and a [Vec3i].
+ */
+public infix fun Vec3f.cross(other: Vec3i): Vec3f {
+    return Vec3f(
+        this.y * other.z - this.z * other.y,
+        this.z * other.x - this.x * other.z,
+        this.x * other.y - this.y * other.x
+    )
+}
+
+/**
+ * The dot product of a [Vec3f] and a [Vector3d].
+ */
+public infix fun Vec3f.dot(other: Vector3d): Float {
+    return this.dot(Vec3f(other.x.toFloat(), other.y.toFloat(), other.z.toFloat()))
+}
+
+/**
+ * The cross product of a [Vec3f] and a [Vector3d].
+ */
+public infix fun Vec3f.cross(other: Vector3d): Vec3f {
+    return Vec3f(
+        (this.y * other.z - this.z * other.y).toFloat(),
+        (this.z * other.x - this.x * other.z).toFloat(),
+        (this.x * other.y - this.y * other.x).toFloat()
+    )
+}
+
+/**
+ * The length of a [Vec3f].
+ */
+public fun Vec3f.length(): Float {
```

Comment:
"magnitude" would be the more mathematically correct term here, isn't it

Reply:
Take it up with the mappings team, haha
I just copied the name from Minecraft source

Reply:
fairs :ioa: